File: GreenMixManager/WebContent/scripts/mqtt.js
/**
 * @author srikanth
 */

$.getScript("scripts/MQTT.js");
var MQttClient = (function(){
	
	return {
		
		pahoClient:null,
		isConnected:false,
		topics:{},
		init:function(obj){
			var thisRef = this;
			this.pahoClient = new Paho.MQTT.Client(obj.host, 
													Number(obj.port), 
													obj.clientId);
			
			this.pahoClient.onConnectionLost = function(message){
				alert(JSON.stringify(message));
				if (message.errorCode !== 0) {
				    alert("onConnectionLost:"+message.errorMessage);
				  }
				thisRef.pahoClient.disconnect();
			};
			this.pahoClient.onMessageArrived = function(data){
				   //alert(data);
				   obj.messageCallBack(data);
			   };
			
			this.pahoClient.connect({userName:obj.user,
									 password:obj.password,
									 onSuccess:function(){
										 		thisRef.isConnected = true;
										 		//thisRef.publish({name:"srikanth"});
										 		thisRef.subscribe();
										 		
										 		
										 		
										 		//alert("connected");
									 			}
								
									 		
									});
			
			
			
		},
		setTopics:function(obj){
			this.topics.subscribeTopic = obj.subscribe;
			this.topics.publishTopic = obj.publish;
		},
		publish:function(obj){
			//if(this.isConnected ){
				var mqttMessage = new Paho.MQTT.Message(JSON.stringify(obj));
				mqttMessage.destinationName = this.topics.publishTopic;
				this.pahoClient.send(mqttMessage);
			//}
		},
	   subscribe:function(){
		   if(this.isConnected){
			  
			   this.pahoClient.subscribe(this.topics.subscribeTopic,
						  
					  	{
				  		qos:0,
				  		onSuccess:function(message){
				  			//alert("subscribed");
				  			//alert("successfully subscribed "+JSON.stringify(message));
				  		},onFailure:function(message){
				  			alert(message);
				  		}
			  
				});
		   }
	   },
	   registerMessageArrivalCallBack:function(fun){
		   this.pahoClient.onMessageArrived = function(data){
			   alert(data);
			   fun(data);
		   };
	   }
	   
		
	};
	
	
	
})();

File: GreenMixManager/docs/greenmix/Mqtt.py
#-*- coding: utf-8 -*-
import paho.mqtt.client as mqtt

#author Srikanth Peresandra Lakshminarayana

class MqttClient:
        def __init__(self,clientId,clean_session,protocol,username,password,host,port):
                self.mqttClient = mqtt.Client(clientId,clean_session,protocol)
                self.mqttUser = username
                self.mqttPassword = password
                self.mqttHost = host
                self.mqttPort = port

        def init(self):
                self.mqttClient.username_pw_set(self.mqttUser,self.mqttPassword)
		#self.mqttClient.connect(self.mqttHost,self.mqttPort)
                #self.mqttClient.on_message = self.onReception
                self.mqttClient.on_connect = self.onConnection
		self.mqttClient.connect(self.mqttHost,self.mqttPort)
        def setTopics(self,publishTopic,subscribeTopic):
                self.publishingTopic = publishTopic
                self.subscriptionTopic = subscribeTopic
        def onConnection(self,client, userdata, flags, rc):
   		print "connected "
                client.subscribe(self.subscriptionTopic)
        def publish(self,data):
                self.mqttClient.publish(self.publishingTopic,data)
        def onReception(self,client, userdata, msg):
                print "received message ",str(msg);
